Specifying absolute `warSourceDirectory` results in error

If `warSourceDirectory` resolves to an absolute pathname, then an exception will be thrown by most liberty goals:

```
Caused by: java.io.IOException: The filename, directory name, or volume label syntax is incorrect
        at java.io.WinNTFileSystem.canonicalize0(Native Method)
        at java.io.WinNTFileSystem.canonicalize(WinNTFileSystem.java:428)
        at java.io.File.getCanonicalPath(File.java:618)
        at io.openliberty.tools.maven.PluginConfigXmlDocument.createElement(PluginConfigXmlDocument.java:54)
        at io.openliberty.tools.maven.server.PluginConfigSupport.exportParametersToXml(PluginConfigSupport.java:171)
        at io.openliberty.tools.maven.server.PluginConfigSupport.installServerAssembly(PluginConfigSupport.java:75)
        at io.openliberty.tools.maven.server.CreateServerMojo.doExecute(CreateServerMojo.java:59)
        at org.codehaus.mojo.pluginsupport.MojoSupport.execute(MojoSupport.java:122)
```

This happens because this:

https://github.com/OpenLiberty/ci.maven/blob/liberty-maven-3.2/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/PluginConfigSupport.java#L353

— Always assumes that `dir` is not absolute (see lines 354 & 356).

`<warSourceDirectory>${basedir}/src/main/webapp`</warSourceDirectory>` is, while redundant, is still considered a valid value as far as the Maven WAR plugin is concerned.

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/PluginConfigSupport.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/PluginConfigSupport.java
@@ -350,11 +350,22 @@ public class PluginConfigSupport extends StartDebugMojoSupport {
 
     protected File getWarSourceDirectory(MavenProject proj) {
         String dir = getPluginConfiguration(proj, "org.apache.maven.plugins", "maven-war-plugin", "warSourceDirectory");
+        log.debug(
+        		String.format("WAR source directory from POM: %s", dir));
+        File warSourceDir;
+        
         if (dir != null) {
-            return new File(proj.getBasedir(), dir);
+        	File dirObj = new File(dir);
+        	if (dirObj.isAbsolute()) {
+        		warSourceDir = dirObj;
+        	} else {
+        		warSourceDir = new File(proj.getBasedir(), dir);
+        	}
         } else {
-            return new File(proj.getBasedir(), "src/main/webapp");
+        	warSourceDir = new File(proj.getBasedir(), "src/main/webapp");
         }
+        log.debug(String.format("Final WAR source directory: %s (absolute: %s)", warSourceDir, warSourceDir.getAbsolutePath()));
+        return warSourceDir;
     }
 
     private String getPluginConfiguration(MavenProject proj, String pluginGroupId, String pluginArtifactId,

